tests/scalers/temporal/temporal_test.go

```diff
@@ -0,0 +1,339 @@
+kind: Deployment
+metadata:
+  name: {{.TemporalDeploymentName}}
+  namespace: {{.TestNamespace}}
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: temporal
+  template:
+    metadata:
+      labels:
+        app: temporal
+    spec:
+      containers:
+      - env:
+        - name: AUTO_SETUP
+          value: "true"
+        - name: DB
+          value: postgresql
+        - name: DB_PORT
+          value: "5432"
+        - name: POSTGRES_USER
+          value: {{.PgRootUserName}}
+        - name: POSTGRES_PWD
+          value: {{.PgRootPassword}}
+        - name: POSTGRES_SEEDS
+          value: {{.PgDeploymentName}}.{{.TestNamespace}}.svc.cluster.local
+        image: temporalio/auto-setup:1.20.1
+        imagePullPolicy: IfNotPresent
+        livenessProbe:
+          exec:
+            command:
+            - /bin/sh
+            - -ec
+            - test $(ps -ef | grep -v grep | grep temporal-server | wc -l) -eq 1
+          failureThreshold: 3
+          initialDelaySeconds: 5
+          periodSeconds: 30
+          successThreshold: 1
+          timeoutSeconds: 1
+        name: temporal
+        ports:
+        - containerPort: 7233
+          protocol: TCP
+        readinessProbe:
+          exec:
+            command:
+            - /bin/sh
+            - -ec
+            - test $(ps -ef | grep -v grep | grep temporal-server | wc -l) -eq 1
+          failureThreshold: 3
+          initialDelaySeconds: 5
+          periodSeconds: 30
+          successThreshold: 1
+          timeoutSeconds: 1
+`
+	scaledObjectTemplate = `
+apiVersion: keda.sh/v1alpha1
+kind: ScaledObject
+metadata:
+  name: {{.ScaledObjectName}}
+  namespace: {{.TestNamespace}}
+spec:
+  scaleTargetRef:
+    name: {{.DeploymentName}}
+  pollingInterval: 5
+  cooldownPeriod:  10
+  minReplicaCount: 0
+  maxReplicaCount: 1
+  advanced:
+    horizontalPodAutoscalerConfig:
+      behavior:
+        scaleDown:
+          stabilizationWindowSeconds: 10
+  triggers:
+  - type: temporal
+    metadata:
+      namespace: default
+      workflowName: SayHello
+      activityName: say_hello
+      targetQueueSize: "2"
+      activationTargetQueueSize: "3"
+      endpoint: {{.TemporalDeploymentName}}.{{.TestNamespace}}.svc.cluster.local:7233
+`
+	deploymentTemplate = `
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{.DeploymentName}}
+  namespace: {{.TestNamespace}}
+  labels:
+    app: {{.DeploymentName}}
+spec:
+  replicas: 0
+  selector:
+    matchLabels:
+      app: {{.DeploymentName}}
+  template:
+    metadata:
+      labels:
+        app: {{.DeploymentName}}
+    spec:
+      containers:
+      - name: nginx
+        image: nginxinc/nginx-unprivileged
+        ports:
+        - containerPort: 80
+`
+	jobWorkerTemplate = `
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: worker
+  namespace: {{.TestNamespace}}
+spec:
+  template:
+    spec:
+      containers:
+      - name: worker
+        image: "prajithp/temporal-sample:1.0.0"
```

Comment:
Could you open a PR to [this repo](https://github.com/kedacore/test-tools) adding the image? We prefer to have all the used images in the org infra to prevent possible issues